client/app/app.js

```diff
@@ -472,66 +472,67 @@ angular
         $translateProvider.use(defaultLanguage);
         $translateProvider.fallbackLanguage("fr_FR");
     })
-    .config(($stateProvider, $httpProvider) => {
+    .config(($transitionsProvider, $httpProvider) => {
         "use strict";
 
         $httpProvider.interceptors.push("translateInterceptor");
 
-        $stateProvider.decorator("translations", (state) => {
-            const routeOption = state.self;
+        $transitionsProvider.onBefore({}, (transition) => {
 
-            if (routeOption.translations) {
+            transition.addResolvable({
+                token: "translations",
+                deps: ["$translate", "$translatePartialLoader"],
+                resolveFn: ($translate, $translatePartialLoader) => {
+                    const state = transition.to();
 
-                const templateUrlTab = [];
-                let translationsTab = routeOption.translations;
+                    if (state.translations) {
 
-                if (routeOption.views) {
-                    angular.forEach(routeOption.views, (value) => {
+                        const templateUrlTab = [];
+                        let translationsTab = state.translations;
 
-                        if (_.isUndefined(value.noTranslations) && !value.noTranslations) {
-                            if (value.templateUrl) {
-                                templateUrlTab.push(value.templateUrl);
-                            }
-                            if (value.translations) {
-                                translationsTab = _.union(translationsTab, value.translations);
-                            }
-                        }
-
-                    });
-                }
+                        if (state.views) {
+                            angular.forEach(state.views, (value) => {
```

Comment:
Why use `angular`, we mostly use `_`

Reply:
It looks like you could have two `map` instead of one `foreach`, maybe it would be more readable